anaconda_linter/lint/__init__.py

```diff
@@ -380,6 +380,45 @@ def check_recipe(self, recipe_name: str, arch_name: str, recipe: RecipeReaderDep
         :param recipe: Recipe to be checked
         """
 
+    def _validate_value(self, value: any) -> bool:  # pylint: disable=unused-argument
+        """
+        checks the value is valid
+
+        :param value: Value to be checked
+        """
+        return False
+
+    def _validate_if_recipe_path_is_missing(
+        self,
+        section_path: str,
+        severity: Severity = SEVERITY_DEFAULT,
+    ) -> None:
+        """
+        Validate if a recipe path is missing
+        Helper function to check if a recipe path is missing.
+
+        :param section_path: Path of the section to be checked
+        :param severity: Severity of the message
+        """
+        recipe = self.recipe
+        if recipe.contains_value(section_path):
+            value = recipe.get_value(section_path)
+            if value is not None and self._validate_value(value):
+                return
+        if not recipe.is_multi_output():
```

Comment:
```suggestion
            self.message(section=section_path, severity=severity)
            return
        if not recipe.is_multi_output():
```
If a path is found at the root-level, but the value is invalid, maybe we shouldn't attempt to examine the outputs ?